### DISCUSSION TITLE
API Channel - Private Message also triggers a webhook POST . Can this be disabled

### DISCUSSION THREAD

**Discussion by devrulabs** [chatwoot/chatwoot#2435] [Q&A] [answered]
is there a way to disable the webhook call for private messages ? 

Currently the private message attribute in the API defines the type of message , however if we use the API channel to build custom integrations and its not always needed that the Agent to Agent Private conversations needs to be sent to the Webhook listener for privacy reasons.

Can we disable the webhook call for private messages between the Agents so the external webhook only receives events related the Conversations between the Agents and the requests originating from the API channel ?

**Comment by sojan-official** [comment, marked as answer]
There isn't a way at the moment. The APIs are built with the idea that the webhook server which receives the messages has admin privileges. It can then filter out the type of messages which you want to ignore and pass on the appropriate information to your clients.

For the time being, you could set up a cloud function or an intermediary server that filters the events. We consider the request for our backlog.
